src/resolve/resolvconf-compat.c

```diff
@@ -268,9 +268,10 @@ int resolvconf_parse_argv(int argc, char *argv[]) {
                 } else if (type == TYPE_PRIVATE)
                         log_debug("Private DNS server data not supported, ignoring.");
 
-                if (!arg_set_dns)
-                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
-                                               "No DNS servers specified, refusing operation.");
+                if (!arg_set_dns && !arg_set_domain) {
+                        log_debug("Neither DNS servers nor domains specified. There is nothing we need to do.");
+                        return 0;
+                }
```

Comment:
but shouldn't this continue to fail if you set search domains but not dns server along with it?

Reply:
Previously, when no domain is specified, the current domains are kept, and only DNS servers are set. That is, only DNS servers can be updated without touching domains. However, we could not update only domains without touching DNS servers. That sounds spurious for me.

Another option is, clear domains when no domain is specified. But, that may break backward compatibility.

Reply:
After reconsidering about this, typical use case of resolvconf is passing resolv.conf generated by a user or program e.g. DHCP client. Hence, the input should be complete.
So, now I picked the another option, that is, clear domains when nothing specified.